addons/website/static/src/js/theme_preview_form.js

```diff
@@ -28,53 +27,46 @@ var ThemePreviewController = FormController.extend({
     /**
      * @override
      */
-    autofocus: function () {
-        // force refresh label of button switch
-        this.$switchButton = this._renderSwitchButton();
-        $('.o_switch_mode_button').replaceWith(this.$switchButton);
-        this._super.apply(this, arguments);
-    },
-     /**
-     * @override
-     */
     renderButtons: function ($node) {
         var $previewButton = $(qweb.render('website.ThemePreview.Buttons'));
         $node.html($previewButton);
-        this.$switchButton = this._renderSwitchButton();
-        $node.find('.o_switch_mode_button').replaceWith(this.$switchButton);
     },
-
     // -------------------------------------------------------------------------
     // Private
     // -------------------------------------------------------------------------
-     /**
-     * Return jQuery Element button 'Switch Mode' with correct labelling.
+    /**
+     * Add Switcher View Mobile / Desktop near pager
      *
      * @private
      */
-    _renderSwitchButton: function () {
-        var isMobile = !!this.$('.is_mobile').length;
-        return $(qweb.render('website.ThemePreview.SwitchModeButton', {
-            'icon': isMobile ? 'fa-desktop' : 'fa-refresh',
-            'PreviewType': isMobile ? _t('Desktop') : _t('Mobile'),
-        }));
+    _updatePager: function () {
+        this._super(...arguments);
+
+        const $buttonSwitch = $(qweb.render('website.ThemePreview.SwitchModeButton'));
+        if (!this.$switcherButton) {
+            this.$switcherButton = $buttonSwitch;
+            this.$switcherButton.appendTo(this.pager.$el);
+        } else {
+            const $switchButton = this.$switcherButton;
+            this.$switcherButton = $buttonSwitch;
+            $switchButton.replaceWith(this.$switcherButton);
+
+        }
```

Comment:
You can remove the extra blank line and you can put the duplicated line `this.$switcherButton = $buttonSwitch;` after the if and else. And inside the if and else, use the `$buttonSwitch` variable. The goal is just to have less lines of code (untested):
```javascript
        const $buttonSwitch = $(qweb.render('website.ThemePreview.SwitchModeButton'));
        if (!this.$switcherButton) {
            $buttonSwitch.appendTo(this.pager.$el);
        } else {
            this.$switcherButton.replaceWith($buttonSwitch);
        }
        this.$switcherButton = $buttonSwitch;
```

Sorry if you find that useless, it is just my opinion, you can disagree.